anchore_engine/services/catalog/api/controllers/default_controller.py

```diff
@@ -95,7 +96,9 @@ def add_image(image_metadata=None, tag=None, digest=None, created_at=None, from_
 
             with db.session_scope() as session:
                 return_object, httpcode = anchore_engine.services.catalog.catalog_impl.image(session, request_inputs, bodycontent=image_metadata)
-
+    except ImageConflict as img_err:
```

Comment:
this is a good way to deal with exceptions... first the specific ones, then the generic ones. Good changes!